feature_engine/variable_handling/_variable_type_checks.py

```diff
@@ -25,15 +26,16 @@ def _is_categories_num(column: pd.Series) -> bool:
 
 
 def _is_convertible_to_dt(column: pd.Series) -> bool:
-    return is_datetime(pd.to_datetime(column, errors="ignore", utc=True))
+    with warnings.catch_warnings():
```

Comment:
which error is this change fixing?

Reply:
In the case when the values of `column` are a string not convertible to datetime, `pandas.to_datetime` would raise a warning saying that it was unable to find the correct date format.

Reply:
For reference, this is the warning that is raised:

```
UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
```

This warning has been introduced in pandas 2.0

Code in pandas 1.5.2
https://github.com/pandas-dev/pandas/blob/1.5.x/pandas/core/tools/datetimes.py#L127

Code in pandas 2.0
https://github.com/pandas-dev/pandas/blob/2.0.x/pandas/core/tools/datetimes.py#L132

Reply:
Thank you! Very helpful!